### PR TITLE
semver checker: do not return error msg on new args with default values

### PR DESCRIPTION
**Pull request by ylil93**
- updates package crawler to only include class/function args that don't have default values

### CODE REVIEW

File: compatibility_lib/compatibility_lib/package_crawler_static.py [hunk 1]
```diff
@@ -219,9 +219,11 @@ def _get_function_info(functions):
 
 
 def _get_args(node):
-    """returns a list of args"""
+    """returns a list of non-default args, ignoring args with defaults"""
     args = []
-    for arg in node.args:
+    num_args = len(node.args) - len(node.defaults)
```
Comment (brianquinlan) [inline compatibility_lib/compatibility_lib/package_crawler_static.py]: Python argument lists can be pretty complex e.g.
def baz(foo, bar=3, *args, key=5): pass

You might want to write a bunch of tests to make sure that you handle all of the edge cases.

Comment (ylil93) [reply compatibility_lib/compatibility_lib/package_crawler_static.py]: Oh nice catch - I forgot about that way of passing args.

Quick question about *args and **kwargs though: would a signature like this ever appear?
`foo(a, *args, c): pass`

Comment (brianquinlan) [reply compatibility_lib/compatibility_lib/package_crawler_static.py]: I don't know how often that appears in practice. I like to do foo(a, *, b, c, d) because I'm a fan of keyword arguments ;-)

Comment (ylil93) [reply compatibility_lib/compatibility_lib/package_crawler_static.py]: i'll add more test cases!

Comment (ylil93) [reply compatibility_lib/compatibility_lib/package_crawler_static.py]: done

File: compatibility_lib/compatibility_lib/semver_checker.py [hunk 2]
```diff
@@ -30,7 +29,7 @@ def check(old_dir, new_dir):
         new_dir: directory containing new files
 
     Returns:
-        False if changes breaks semver, True if semver is preserved
+        a list of error strings describing semver breakages
```
[the remark this answers is not in the document]
Comment (ylil93) [reply compatibility_lib/compatibility_lib/semver_checker.py]: done

File: compatibility_lib/compatibility_lib/package_crawler_static.py [hunk 3]
```diff
@@ -63,9 +63,14 @@ def get_package_info(root_dir):
             module_name: {
                 'classes': {
                     class_name: {
-                        'args': [arg1, arg2, ...],
+                        'args': {
+                            'single_args': [arg1, arg2, ...],
+                            'defaults': {arg1: value1, ...},
+                            'vararg': argv,
+                            'kwarg': kwargs,
```
Comment (brianquinlan) [inline compatibility_lib/compatibility_lib/package_crawler_static.py]: Don't you need to know the position of these things?

Comment (ylil93) [reply compatibility_lib/compatibility_lib/package_crawler_static.py]: I get the args using ast tooling which returns the args in order.
